File: clients/node/src/admin/jobs.ts
import { EndpointGroup } from '../base.js'

export class AdminJobsAPI extends EndpointGroup {
  run(payload: { coid: string; job_type: string; processing_mode?: string; priority?: boolean }) {
    return this.requestPost<Record<string, unknown>>('/api/admin/jobs/run', {
      json: {
        coid: payload.coid,
        jobType: payload.job_type,
        ...(payload.processing_mode ? { processingMode: payload.processing_mode } : {}),
        priority: payload.priority ?? false,
      },
    })
  }

  list(
    coid: string,
    params?: { limit?: number; offset?: number }
  ) {
    return this.requestGet<Record<string, unknown>>(`/api/admin/jobs/${coid}`, {
      params:
        params && (params.limit != null || params.offset != null)
          ? {
              limit: params.limit,
              offset: params.offset,
            }
          : undefined,
    })
  }

  cancel(payload: { coid: string; dag_run_id: string }) {
    return this.requestPost<Record<string, unknown>>(
      `/api/admin/jobs/${payload.coid}/${payload.dag_run_id}/cancel`
    )
  }
}

File: clients/python/src/cien_agent_sdk/admin/jobs.py
from __future__ import annotations

from typing import Any

from ..base import EndpointGroup


class AdminJobsAPI(EndpointGroup):
    """/api/admin/jobs endpoints."""

    def run(self, *, coid: str, job_type: str, processing_mode: str | None = None, priority: bool = False) -> dict[str, Any]:
        """Trigger a job for one company."""
        return self._post(
            "/api/admin/jobs/run",
            json={
                "coid": coid,
                "jobType": job_type,
                **({"processingMode": processing_mode} if processing_mode is not None else {}),
                "priority": priority,
            },
        )

    def list(self, coid: str, *, limit: int | None = None) -> list[dict[str, Any]]:
        """List jobs for one company."""
        params = {"limit": limit} if limit is not None else None
        return self._get(f"/api/admin/jobs/{coid}", params=params)

    def cancel(self, *, coid: str, dag_run_id: str) -> dict[str, Any]:
        """Cancel one dag run for one company."""
        return self._post(f"/api/admin/jobs/{coid}/{dag_run_id}/cancel")
